weavedns/main.go

```diff
@@ -64,7 +64,12 @@ func main() {
 	}
 
 	go weavedns.ListenHttp(weavedns.LOCAL_DOMAIN, zone, httpPort)
-	err := weavedns.StartServer(zone, iface, dnsPort, wait)
+	srv, err := weavedns.NewDNSServer(zone, iface, dnsPort)
+	if err != nil {
+		Error.Fatal("Failed to start server", err)
+	}
+
+	srv.Start()
```

Comment:
presumably you meant to write `err := srv.Start()`, given the code that follows.